Framework/API/src/ExperimentInfo.cpp

```diff
@@ -270,19 +273,24 @@ void ExperimentInfo::makeAPIComponentInfo(const InfoComponentVisitor &visitor) {
 /**
  * Take the visitor from the instrument if it's already there or make from
  * scratch.
- * @param instrument : Instrument which might carry a visitor.
+ * @param parInstrument : Parameterised instrument.
+ * @param newInstrument : unparameterised new instrument
```

Comment:
Probably. But ExperimentInfo::setInstrument also supports arguments where the Instrument is not parameterised. `InfoComponentVisitor` needs a ParameterMap, which it purges of information as it parses through the instrument tree.